content/blog/rsample-1-1-0/index.Rmd

```diff
@@ -0,0 +1,161 @@
+
+The other big change to grouped resampling comes as a new argument to `group_vfold_cv()`. By default, `group_vfold_cv()` assigns roughly the same number of groups to each of your folds, so you wind up with the same number of patients, or neighborhoods, or whatever else you're grouping by in each assessment set. The new `balance` argument lets you instead assign roughly the same number of rows to each fold instead, if you set `balance = observations`:
+
+```{r}
+group_vfold_cv(ames, Neighborhood, balance = "observations")
+```
+
+This approach works in a similar way to the new grouped resampling functions, attempting to assign roughly `1 / v` of your data to each fold. When working with unbalanced groups, this can result in much more even assignments of data to each fold:
+
+```{r}
+library(ggplot2)
+library(dplyr)
+
+analysis_sd <- function(v, balance) {
+  group_vfold_cv(
+    ames, 
+    Neighborhood, 
+    v, 
+    balance = balance
+  )$splits %>% 
+    purrr::map_dbl(~ nrow(analysis(.x))) %>% 
+    sd()
+}
+
+resample <- tidyr::crossing(
+  idx = seq_len(100),
+  v = c(2, 5, 10, 15),
+  balance = c("groups", "observations")
+)
+
+resample %>%
+  mutate(sd = purrr::pmap_dbl(
+    list(v, balance),
+    analysis_sd
+  )) %>%
+  ggplot(aes(sd, fill = balance)) + 
+  geom_histogram(alpha = 0.6, color = "black", size = 0.3) + 
+  facet_wrap(~ v) + 
+  theme_minimal() + 
+  labs(title = "sd() of nrow(analysis) by balance method")
+```
+
+Right now, these grouping functions don't support stratification. If you have thoughts on how you'd expect stratification to work with grouping, or know of examples of other implementations, [let us know on GitHub](https://github.com/tidymodels/rsample/issues/317)!
+
+## Other Improvements
+
+In addition, this release adds a few new utility functions to make it easier to work with the rsets produced by rsample functions.
+
+For instance, the new `reshuffle_rset()` will re-generate an rset, using the same arguments as were used to originally create it, but under the current random seed:
+
+```{r}
+resample <- vfold_cv(ames)
+
+resample$splits[[1]] %>%
+  analysis() %>%
+  head()
+
+resample <- reshuffle_rset(resample)
+
+resample$splits[[1]] %>%
+  analysis() %>%
+  head()
+```
+
+This works with repeated cross-validation, stratification, grouping -- pretty much anything you did originally will be preserved when reshuffling the rset.
```

Comment:
Is there a good brief description of what _won't_ be preserved?

Reply:
Anything that isn't preserved is a bug :sweat_smile: Should I cut the `pretty much`?

Reply:
ha, yeah, that is what prompted me to ask that question 😆

Reply:
Alright, took out "pretty much" (but did replace "will" with "should" to keep a similar hedge :laughing:)